src/test/java/com/cst438/controller/AssignmentControllerUnitTest.java

```diff
@@ -0,0 +1,157 @@
+                    .getResponse();
+
+            // Create user for enterFinalGradesSuccess
+            UserDTO student3 = new UserDTO(0, "Student Three", "student3@example.com", "STUDENT");
+            mvc.perform(
+                            MockMvcRequestBuilders
+                                    .post("/users")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content(asJsonString(student3)))
+                    .andReturn()
+                    .getResponse();
+
+            setUpIsDone = true;
+        }
+    }
+
+    @Test
+    public void gradeAssignmentSuccess() throws Exception {
+        MockHttpServletResponse response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .get("/assignments/1/grades")
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn()
+                .getResponse();
+
+        assertEquals(200, response.getStatus(), "Expected status 200 but got " + response.getStatus());
+
+        GradeDTO[] grades = fromJsonString(response.getContentAsString(), GradeDTO[].class);
+        GradeDTO[] updatedGrades = new GradeDTO[grades.length];
+        for (int i = 0; i < grades.length; i++) {
+            GradeDTO grade = grades[i];
+            updatedGrades[i] = new GradeDTO(grade.gradeId(), grade.studentName(), grade.studentEmail(), grade.assignmentTitle(), grade.courseId(), grade.sectionId(), grade.score() + 5); // Example modification
+        }
+
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .put("/grades")
+                                .content(asJsonString(updatedGrades))
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn()
+                .getResponse();
+
+        assertEquals(200, response.getStatus(), "Expected status 200 but got " + response.getStatus());
+    }
+
+    @Test
+    public void gradeAssignmentInvalidAssignmentId() throws Exception {
+        MockHttpServletResponse response;
+
+        List<GradeDTO> grades = List.of(
+                new GradeDTO(9999, "Student One", "student1@example.com", "Assignment One", "CST438", 1, 95),
+                new GradeDTO(9999, "Student Two", "student2@example.com", "Assignment One", "CST438", 1, 85)
+        );
+
+        response = mvc.perform(
+                        MockMvcRequestBuilders
+                                .put("/grades")
+                                .content(asJsonString(grades))
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn()
+                .getResponse();
+
+        assertEquals(404, response.getStatus());
+    }
+
+    @Test
+    public void enterFinalGradesSuccess() throws Exception {
+        MockHttpServletResponse response;
+
+        List<EnrollmentDTO> finalGrades = List.of(
```

Comment:
Rather than creating a list of EnrollmentDTOs artificially, the rubric says to invoke REST api GET for the url /sections/{sectionNo}/enrollments. You then update the returned list of EnrollmentDTOs, then PUT them as you do below.

Reply:
Looks good now.